app/src/main/java/org/jellyfin/androidtv/preference/DisplayPreferencesStore.kt

```diff
@@ -71,47 +71,59 @@ abstract class DisplayPreferencesStore(
-		Boolean::class -> cachedPreferences[preference.key]?.toBooleanStrictOrNull()
-			?: preference.defaultValue
-		String::class -> cachedPreferences[preference.key] ?: preference.defaultValue
-
-		else -> throw IllegalArgumentException("${preference.type.simpleName} type is not supported")
-	} as V
-
-	override fun <T : Preference<V>, V : Any> set(preference: T, value: V) = when (preference.type) {
-		Int::class -> cachedPreferences[preference.key] = (value as Int).toString()
-		Long::class -> cachedPreferences[preference.key] = (value as Long).toString()
-		Boolean::class -> cachedPreferences[preference.key] = (value as Boolean).toString()
-		String::class -> cachedPreferences[preference.key] = (value as String).toString()
-		Enum::class -> cachedPreferences[preference.key] = value.toString()
-
-		else -> throw IllegalArgumentException("${preference.type.simpleName} type is not supported")
+	override fun getInt(key: String, defaultValue: Int) =
+		cachedPreferences[key]?.toIntOrNull() ?: defaultValue
+
+	override fun getLong(key: String, defaultValue: Long) =
+		cachedPreferences[key]?.toLongOrNull() ?: defaultValue
+
+	override fun getBool(key: String, defaultValue: Boolean) =
+		cachedPreferences[key]?.toBooleanStrictOrNull() ?: defaultValue
+
+	override fun getString(key: String, defaultValue: String) =
+		cachedPreferences[key] ?: defaultValue
+
+	override fun setInt(key: String, value: Int) {
+		cachedPreferences[key] = value.toString()
 	}
 
-	override fun <T : Preference<V>, V : Enum<V>> get(preference: T): V {
-		val stringValue = cachedPreferences[preference.key]
+	override fun setLong(key: String, value: Long) {
+		cachedPreferences[key] = value.toString()
+	}
 
-		return if (stringValue.isNullOrBlank()) preference.defaultValue
-		else preference.type.java.enumConstants?.find {
-			(it is PreferenceEnum && it.serializedName == stringValue) || it.name == stringValue
-		} ?: preference.defaultValue
+	override fun setBool(key: String, value: Boolean) {
+		cachedPreferences[key] = value.toString()
 	}
 
-	override fun <T : Preference<V>, V : Enum<V>> set(preference: T, value: V) {
-		cachedPreferences[preference.key] = when (value) {
-			is PreferenceEnum -> value.serializedName
-			else -> value.toString()
-		}
+	override fun setString(key: String, value: String) {
+		cachedPreferences[key] = value
 	}
 
-	override fun <T : Preference<V>, V : Any> delete(preference: T) {
+	override fun <T : Any> delete(preference: Preference<T>) {
 		cachedPreferences.remove(preference.key)
 	}
 
+	override fun <T : Enum<T>> getEnum(preference: Preference<T>): T {
+		val stringValue = getString(preference.key, "")
+
+		if (stringValue.isBlank()) {
+			return preference.defaultValue
+		}
+
+		val loadedVal = preference.type.java.enumConstants?.find {
+			(it is PreferenceEnum && it.serializedName == stringValue) || it.name == stringValue
+		} ?: preference.defaultValue
+		return loadedVal
```

Comment:
Not entirely sure why you changed the implementation, but using null as default is faster as it doesn't allocate an empty string, it also should not use the getString() function but read from the cached data directly.

(same for the implementation in SharedPrefs)
```suggestion
		val stringValue = cachedPreferences[preference.key]

		return if (stringValue.isNullOrBlank()) preference.defaultValue
		else preference.type.java.enumConstants?.find {
			(it is PreferenceEnum && it.serializedName == stringValue) || it.name == stringValue
		} ?: preference.defaultValue
```

Reply:
In `SharedPreferenceStore` I've opted for the following:
```
		val stringValue = getString(preference.key, "")
		return if (stringValue.isBlank()) preference.defaultValue
		// <unmodified>
```

This is so all callers of `getString` don't have to handle the null case which is an implementation detail for `getEnum`. For the shared store I'm tempted to leave it, it's a small performance pessimism but avoids us having to put null checks everywhere when we (de)serialise strings elsewhere